crates/runtime/src/dataconnector/listing/connector.rs

```diff
@@ -331,6 +333,148 @@ pub trait ListingTableConnector: DataConnector {
             source: error.into(),
         }
     }
+
+    async fn create_text_table(
+        &self,
+        dataset: &Dataset,
+        url: &Url,
+        extension: &str,
+    ) -> DataConnectorResult<Arc<dyn TableProvider>>
+    where
+        Self: Display,
+    {
+        let content_formatter =
+            document_parse::get_parser_factory(extension)
+                .await
+                .map(|factory| {
+                    // TODO: add opts.
+                    factory.default()
+                });
+
+        Ok(ObjectStoreTextTable::try_new(
+            self.get_object_store(dataset)?,
+            &url.clone(),
+            Some(extension.to_string()),
+            content_formatter,
+        )
+        .context(crate::dataconnector::InvalidConfigurationSnafu {
+            dataconnector: format!("{self}"),
+            connector_component: ConnectorComponent::from(dataset),
+            message: format!(
+                "Invalid file extension ({extension}) for source ({})",
+                dataset.name
+            ),
+        })?)
+    }
+
+    async fn create_listing_table(
+        &self,
+        dataset: &Dataset,
+        url: &Url,
+        extension: &str,
+        file_format: Arc<dyn FileFormat>,
+    ) -> DataConnectorResult<Arc<dyn TableProvider>>
+    where
+        Self: Display,
+    {
+        // This shouldn't error because we've already validated the URL in `get_object_store_url`.
+        let table_path = ListingTableUrl::parse(url.clone()).boxed().context(
+            crate::dataconnector::InternalSnafu {
+                dataconnector: format!("{self}"),
+                connector_component: ConnectorComponent::from(dataset),
+                code: "LTC-RP-LTUP".to_string(), // ListingTableConnector-ReadProvider-ListingTableUrlParse
+            },
+        )?;
+
+        let object_store = self.get_object_store(dataset)?;
+
+        let ctx: SessionContext = Self::get_session_context();
+
+        // Get the last modified object for the provided ObjectStore to infer the schema.
+        // Report an error if no files matching required extension are found.
+        let last_modified_or_added = get_last_modified(
```

Comment:
We can use `table_path.is_collection()` to detect whether the path is a folder or a file, and avoid the logic for getting the last modified file. I'm keeping this logic as it does not add overhead and makes the code more readable by reducing if/else branching.